packages/aws_common/test/http/http_payload_test.dart

```diff
@@ -82,5 +84,35 @@ void main() {
         ]),
       );
     });
+
+    group('dataUrl', () {
```

Comment:
Can you add more tests? There are some more examples on these pages:

- https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs
- https://en.wikipedia.org/wiki/Data_URI_scheme